docs/deploy-rbd.md

```diff
@@ -213,6 +218,36 @@ secrets under `encryptionPassphrase` key and switch `encrypted` option in
 StorageClass to `"true"`. This is not supported for storage classes that already
 have PVs provisioned.
 
+### Encryption KMS configuration
+
+To further improve security robustness it is possible to use unique passphrases
+generated for each volume and stored in a Key Management System (KMS). Currently
+HashiCorp Vault is the only KMS supported.
+
+To use Vault as KMS set `encryptionKMS` to `vault` and `encryptionKMSID` to a
+unique identifier for Vault configuration. You will also need to create vault
+configuration similar to the [example](../examples/rbd/kms-config.yaml)
+and use same `encryptionKMSID`. In order for ceph-csi to be able to access the
+configuration you will need to have it mounted to csi-rbdplugin containers as
+`ceph-csi-encryption-kms-config` config map.
+
+#### Configuring HashiCorp Vault
+
+Using Vault as KMS you need to configure Kubernetes authentication method as
+described in [official
+documentation](https://www.vaultproject.io/docs/auth/kubernetes.html).
+
+If token reviewer is used, you will need to configure service account for
+that also like in [example](../examples/rbd/csi-vaulttokenreview-rbac.yaml) to
+be able to review jwt tokens.
+
+Configure a role(s) for service accounts used for ceph-csi:
+
+* provisioner service account (`rbd-csi-provisioner`) requires only **delete**
```

Comment:
why delete permission is given to provisioner here, in that case, both create and delete should be performed by provisioner or both create and delete should be performed by node plugin

@ShyamsundarR any comments here?

Reply:
the question because am not sure you will get network access to  reach vault server from both provisioner and nodeplugin

Reply:
so regarding this:
- volumes are deleted by provisioner, hence delete permissions in vault is required here
- since we're not encrypting volume on create but rather on first attach - we'll need to have create permissions and also read since each attach call will need to get passphrase

considering that in vault you can have quite flexible permissions and have different roles for different actions - it's easy to restrict only to operations that are required

Reply:
I am fine with the reasoning (IOW, Create is by the nodeplugin, and Delete is by the Provisioner).


@Madhu-1 what is the case where vault access from nodeplugin would be prevented? As the provisioner can (like the nodeplugin) run on any node, and so if there are node specific access restrictions to vault, then even the provisioner can face the same, when running on any such worker node, afaict.